ddl/ddl_worker.go

```diff
@@ -542,10 +545,30 @@ func (w *worker) runDDLJob(d *ddlCtx, t *meta.Meta, job *model.Job) (ver int64,
 
 		job.Error = toTError(err)
 		job.ErrorCount++
+		// Load global ddl variables.
+		if err1 := loadDDLVars(w); err1 != nil {
+			log.Errorf("[ddl-%s] load ddl global variable error: %v", w, err1)
+		}
+		// Check error limit to avoid falling into an infinite loop.
+		if job.ErrorCount > variable.GetDDLErrorCountLimit() && job.State == model.JobStateRunning && admin.IsJobRollbackable(job) {
+			log.Warnf("[ddl-%s] the job id %v error count exceed the limit, cancelling it now", w, job.ID)
```

Comment:
Could we add the value of limit  to log?

Reply:
done.